yarl/_url.py

```diff
@@ -1116,7 +1122,59 @@ def join(self, url):
         # See docs for urllib.parse.urljoin
         if not isinstance(url, URL):
             raise TypeError("url should be URL")
-        return URL(urljoin(str(self), str(url)), encoded=True)
+        other: URL = url
+        scheme = other.scheme or self.scheme
+        parts = {
+            k[4:]: getattr(self, k) or ""
+            for k in ("raw_authority", "raw_path", "raw_query_string", "raw_fragment")
+        }
+        parts["scheme"] = scheme
+
+        if scheme != self.scheme or scheme not in uses_relative:
+            return URL(str(other))
```

Comment:
```suggestion
            return URL(other._val._replace(), encoded=True)
```

something like this should work to copy without having to go through `__str__`

Reply:
yes